source/1_engine/rex_engine/src/string/stringentry.cpp

```diff
@@ -0,0 +1,97 @@
+#include "rex_engine/string/stringentry.h"
+
+#include <rex_std/memory.h>
+
+namespace rex
+{
+  //-------------------------------------------------------------------------
+  StringEntry::StringEntry(rsl::string_view characters) noexcept
+      : m_characters()
+  {
+    m_characters.reset(static_cast<char*>(malloc(characters.size() + 1))); // NOLINT(cppcoreguidelines-no-malloc)
+
+    rsl::memset(m_characters.get(), 0, characters.size() + 1);
```

Comment:
why do you set to 0 first, only to copy in it afterwards?

Reply:
Cuz it's zero terminated, IIRC it's faster to clear everything to 0 and copy the data you want over first copy the string and place the 0 afterwards, also I think this looks cleaner.

Reply:
can you put a comment why you do that then?
I'm not sure if it's faster, it'll probably depend on the length of the string.
it'll also depend if this first access to the memory, on first access and if the memory used for this was on a previous unused page, the OS needs to actually initialize the page, that might be what you've seen before.

I'd like to see this profiled in possible though, if you copy everything first and then add a null termination char at the end, I dont' see how this would be slower over what's implemented here

Reply:
setting only the last char is always faster
https://github.com/New-Creta/Rex/issues/103